tests/performance/helpers/horizon-helpers.js

```diff
@@ -0,0 +1,81 @@
+import {
+  sendRequest,
+  getRandom,
+  getAxisInDimension,
+} from "./request-helpers.js";
+
+export function sendHorizonRequest(
+  horizon, surface, above, below, stepsize, attributes
+) {
+  const vds = __ENV.VDS;
+  const sas = __ENV.SAS;
+  let properties = {
+    fillValue: -999.25,
+    horizon: horizon,
+    interpolation: "linear",
+    vds: vds,
+    sas: sas,
+    above: below,
+    below: above,
+    stepsize: stepsize,
+    attributes: attributes
+  };
+  let payload = Object.assign({}, properties, surface);
+  sendRequest("horizon", payload);
+}
+
+export function retrieveMetadataForHorizonRequest(metadata) {
+  const inlineAxis = getAxisInDimension(metadata, "Inline");
+  const xlineAxis = getAxisInDimension(metadata, "Crossline");
+  const samplesAxis = getAxisInDimension(metadata, "Sample");
+
+  const cdp = metadata.boundingBox.cdp;
+
+  const ilineStepX = (cdp[1][0] - cdp[0][0]) / (inlineAxis.samples - 1);
+  const ilineStepY = (cdp[1][1] - cdp[0][1]) / (inlineAxis.samples - 1);
+  const xlineStepX = (cdp[3][0] - cdp[0][0]) / (xlineAxis.samples - 1);
+  const xlineStepY = (cdp[3][1] - cdp[0][1]) / (xlineAxis.samples - 1);
+
+  const rotation = (Math.atan2(ilineStepY, ilineStepX) * 180) / Math.PI;
+  const xinc = Math.hypot(ilineStepX, ilineStepY);
+  const yinc = Math.hypot(xlineStepX, xlineStepY);
+
+  let surface = {
+    xinc: xinc,
+    yinc: yinc,
+    xori: cdp[0][0],
+    yori: cdp[0][1],
+    rotation: rotation,
+  };
+  return [surface, inlineAxis, xlineAxis, samplesAxis];
+}
+
+/**
+ * Sends horizon request of size rows/columns where every point is at provided depth.
+ */
+export function sendConstantHorizonRequest(
+  rows, columns, depth, surface, above, below, stepsize, attributes
+){
+  let rowArray = new Array(columns).fill(depth);
+  let horizon = new Array(rows).fill(rowArray);
+  console.log(`Requesting horizon of size ${columns * rows}`);
+  return sendHorizonRequest(horizon, surface, above, below, stepsize, attributes);
+}
+
+/**
+ * Sends horizon request of size rows/columns where every point is at random depth.
+ * Every point is expected to have data without need for interpolation in any dimension.
+ */
+export function sendRandomHorizonRequest(
```

Comment:
Randomness in a performance test, sure that is a good idea?

Reply:
Absolutely.
I don't care what depth is requested as long as I can assure that
1) we are not likely to hit cache
2) we do not request flat-ish surface and end up downloading too little data

Point 2 might be good to have if that is a real user pattern, but as I am in the process of losing faith in real tests ability to catch degrading performance, there seem to be no point in making it more complicated then necessary.